pkgs/games/ironwail/default.nix

```diff
@@ -0,0 +1,96 @@
+{ lib
+, stdenv
+, SDL2
+, fetchurl
+, gzip
+, libvorbis
+, libmad
+, flac
+, libopus
+, opusfile
+, libogg
+, curl
+, libxmp
+, vulkan-headers
+, vulkan-loader
+, copyDesktopItems
+, makeDesktopItem
+, pkg-config
+}:
+
+stdenv.mkDerivation rec {
```

Comment:
Here, I recommend using the `finalAttrs` pattern. To learn more about it, I suggest referring to the following resources:

1. Historical Discussion: The discourse thread at https://discourse.nixos.org/t/avoid-rec-expresions-in-nixpkgs/8293 provides insights into the benefits of avoiding recursive expressions in Nixpkgs.
2. Documentation: The Nixpkgs manual provides detailed information on `mkDerivation` and recursive attributes. You can find it at https://nixos.org/manual/nixpkgs/unstable/#mkderivation-recursive-attributes.
3. Implementation Example: https://github.com/NixOS/nixpkgs/compare/17f96f7b978e61576cfe16136eb418f74fab9952..9e6ea843e473d34d4f379b8b0d8ef0425a06defe.

I hope these resources provide valuable insights and guidance for your needs. If you have any further questions, feel free to ask.